include/powsybl/iidm/ShuntCompensatorLinearModelAdder.hpp

```diff
@@ -0,0 +1,55 @@
+/**
+ * Copyright (c) 2020, RTE (http://www.rte-france.com)
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+
+#ifndef POWSYBL_IIDM_SHUNTCOMPENSATORLINEARMODELADDER_HPP
+#define POWSYBL_IIDM_SHUNTCOMPENSATORLINEARMODELADDER_HPP
+
+#include <powsybl/iidm/ShuntCompensatorModelAdder.hpp>
+#include <powsybl/stdcxx/math.hpp>
+#include <powsybl/stdcxx/optional.hpp>
+
+namespace powsybl {
+
+namespace iidm {
+
+class ShuntCompensatorAdder;
+
+namespace shunt_compensator {
+
+class ShuntCompensatorLinearModelAdder : public ShuntCompensatorModelAdder {
+public:  // ShuntCompensatorModelAdder
+    ShuntCompensatorAdder& add() override;
+
+public:
+    explicit ShuntCompensatorLinearModelAdder(ShuntCompensatorAdder& parent);
+
+    ShuntCompensatorLinearModelAdder& setBPerSection(double bPerSection);
+
+    ShuntCompensatorLinearModelAdder& setGPerSection(double gPerSection);
+
+    ShuntCompensatorLinearModelAdder& setMaximumSectionCount(unsigned long maximumSectionCount);
+
+private:  // ShuntCompensatorModelAdder
+    std::unique_ptr<ShuntCompensatorModel> build(ShuntCompensator& shuntCompensator, unsigned long sectionCount) const override;
```

Comment:
Agree, but the shunt is not attached yet when calling build (because build may fail, and we don't want to add shunt if ShuntCompensatorModel cannot be created), so it's not possible to call `Shunt::getSectionCount()`

Reply:
I don't understand why you cannot call `Shunt::getSectionCount`: the shunt has been created (but not attached to the network). The shunt is passed as a parameter of this method, so what I miss?

Reply:
The sectionCount depends on the variant, so my comment is not relevant.